Makefile.include

```diff
@@ -516,12 +516,12 @@ BUILDDEPS += $(APPDEPS)
 # Save value to verify it is not modified later
 _BASELIBS_VALUE_BEFORE_USAGE := $(BASELIBS)
 
-# Linker rule
+# Linker rule creating a randomized binary through link time reordering
 $(ELFFILE): FORCE
 ifeq ($(BUILDOSXNATIVE),1)
-  _LINK = $(if $(CPPMIX),$(LINKXX),$(LINK)) $(UNDEF) $$(find $(BASELIBS) -size +8c) $(LINKFLAGS) $(LINKFLAGPREFIX)-no_pie
+  _LINK = $(if $(CPPMIX),$(LINKXX),$(LINK)) $(UNDEF) $$(find $(BASELIBS) -size +8c | $(RIOTTOOLS)/randsort/randsort.py) $(LINKFLAGS) $(LINKFLAGPREFIX)-no_pie
```

Comment:
I'd prefer to factor out the whole ```$$(find ...)```. Then, optionally, randomize it. Then use it here.

This'll have a similar (negative) effect on CI compilation, reproducible builds and test caching.

Reply:
My reasoning behind this was as follows:

1. `find(1)` is used on OS X anyhow.
2. If the randomization sorting tool should be able to handle spaces in file names in most read from standard input anyhow to prevent issues with word splitting.
3. Adding another if/else to the `$(ELFFILE)` target would make the code even more unreadable.

The only suitable alternative I can think of currently is adding another wrapper script which can be used as `$(LINK)` / `$(LINKXX)` and calls `randsort` internally.

Reply:
>     1. `find(1)` is used on OS X anyhow.

Using find is fine. I meant, factor out into an intermediate variable that is then used by `_LINK`.